src/herder/TransactionQueue.cpp

```diff
@@ -266,6 +270,7 @@ isDuplicateTx(TransactionFrameBasePtr oldTx, TransactionFrameBasePtr newTx)
 bool
 TransactionQueue::sourceAccountPending(AccountID const& accountID) const
 {
+    std::lock_guard<std::recursive_mutex> guard(mTxQueueMutex);
```

Comment:
Any insights into lock contention with the current implementation of tx queue synchronization? Might be useful to turn these into RECURSIVE_LOCK_GUARD, so you can observe thread blocking in Tracy.

Reply:
I switched over the mutex I'm using now to be a tracy-compatible one.